cmd/cdk/cdk.go

```diff
@@ -450,33 +483,35 @@ func watchNewLogs(ctx context.Context, rpcClient *ethclient.Client, filter ether
 // - logs are parsed using the contractInstance and contractABI
 // - logs are printed in JSON format
 // - if the log cannot be parsed, the log is printed as is
-func mustPrintLogs(logs []types.Log, contractInstance reflect.Value, contractABI *abi.ABI) {
-	eventsFound := false
+func mustPrintLogs(logs []types.Log, contractInstance reflect.Value, contractABI *abi.ABI) uint {
+	logsPrinted := uint(0)
 	for _, l := range logs {
 		e, _ := contractABI.EventByID(l.Topics[0])
 		if e == nil {
+			logsPrinted++
 			mustPrintJSONIndent(l)
 			continue
 		}
-		eventsFound = true
 
 		var parsedEvent any
 		parseLogMethod, methodFound := contractInstance.Type().MethodByName(fmt.Sprintf("Parse%s", e.Name))
 		if !methodFound {
-			log.Warn().Msgf("Method Parse%s not found", e.Name)
+			log.Warn().Msgf("Method Parse%s not found, printing raw log", e.Name)
+			logsPrinted++
+			mustPrintJSONIndent(l)
 		} else {
 			parsedLogValues := parseLogMethod.Func.Call([]reflect.Value{contractInstance, reflect.ValueOf(l)})
-			parsedEventValue := parsedLogValues[0].Interface()
 			errValue := parsedLogValues[1].Interface()
 			if errValue != nil {
 				log.Warn().Err(errValue.(error)).Msgf("Error parsing log %v", l)
 			} else {
-				parsedEvent = parsedEventValue
+				parsedEvent = parsedLogValues[0].Interface()
```

Comment:
Even if it's pretty clear that `parsedLogValues` would have a length, I think a `len` check isn't a bad idea.